### DISCUSSION TITLE
Customize bus number

### DISCUSSION THREAD

**Discussion by TKone7** [jpcornil-git/HA-mcp23017#36] [Ideas]
Hey, thanks a lot for this custom component. I realized that `custom_components/mcp23017/const.py` defines a constant number for the bus being used `/dev/i2c-#`. In my case this is not sufficient and I would think it would be best to let this be configured.

Would you accept a PR that makes this a config?

**Comment by jpcornil-git** [comment]
Sure/happy to support, just make sure it is optional, support config flow (configuration.yaml not a must), defaults to current bus and doesn't break existing configflow/configuration.yaml configurations.

There are two options wrt bus number configuration:
- it remains a global settings, i.e. just an option to change the current default (low hanging fruit)
- it becomes a component property (better/more complex change but it should only affect the toplevel logic as MCP23017 class has been prepared to support that already)

**Comment by piotrlochowski** [comment]
Please take a look at PR: https://github.com/jpcornil-git/HA-mcp23017/pull/35
Does it solve reported issue?

**Comment by jpcornil-git** [comment]
Thanks for your PR, it looks good/started review

**Comment by jpcornil-git** [comment]
Once [f7971e1](https://github.com/jpcornil-git/HA-mcp23017/commit/f7971e171420ec9fbd7d879abbdfbc76fa2aff2b) tested OK from you side, I'll publish a new 1.3.0 release to make it available to everyone (HACS will pick it up as well)

**Comment by TKone7** [comment]
Hey, thanks so much for tackling this issue. I just deployed this version on my home assistant running on a RPi 4 and it works great. Means I can now remove my patch 🌈 .

Note: I did not run the migration, since I configure through .yaml.

**Comment by jpcornil-git** [comment]
Even with .yaml should go thru the migration code, i.e. if you look at files below, you should see for mcp23017 instances IDs with bus information prefix:
- .storage/core.config_entries
- .storage/core.device_registry
- .storage/core.entity_registry

Cheers,
jpc

**Reply by TKone7** [reply]
Ah yes you are right. Not very familiar with HA integrations under the hood. But I can see the files were updated.

**Comment by jpcornil-git** [comment]
I just released v1.3.0
